pkg/controller/mysql/grant/reconciler.go

```diff
@@ -206,19 +214,30 @@ func (c *external) getPrivileges(ctx context.Context, username, dbname string) (
 		if err := rows.Scan(&grant); err != nil {
 			return nil, nil, errors.Wrap(err, errCurrentGrant)
 		}
-		p := parseGrant(grant, dbname)
-		if p != nil && privileges != nil {
-			// Found more than one grant for this user/DB pair.
-			// This probably shouldn't happen because MySQL groups privileges
-			// for the same user/DB pair in a single grant.
-			// In any case we want to update and ensure the privileges of our grant.
-			return nil, &managed.ExternalObservation{
-				ResourceExists:   true,
-				ResourceUpToDate: false,
-			}, nil
+		p := parseGrant(grant, dbname, table)
+
+		/*
```

Comment:
In my opinion this block is not needed at all. Suppose there are two rows found, even when we send an `ResourceUpToDate: false` there is still absolutely no guarantee that we will end up in desired state. 
At least i fail to see the value added and unfortunately this defensive logic makes it impossible here to use tables. If we use tables we will always have multiple rows in the result (don't see a way to filter them upfront) and should pick the first combination that matches as there should be no other matches of interest.
Any thoughts? :)

Reply:
Your reasoning makes sense, feel free to remove it.

Reply:
removed

Reply:
removed